docs/content/en/docs/concepts/executable-playbooks.md

```diff
@@ -0,0 +1,387 @@
+
+The `parallel` step allows executing multiple branches (in parallel) specified in the `next_steps` field. At this time, the steps in `next_steps` are executed sequentially. Parallel execution is scheduled for a later release in SOARCA.
+
+Next, we explain variables in CACAO and SOARCA.
+
+### Variables
+
+The CACAO specification allows defining variables on the playbook level, as well as on the step level. Playbook variables are available throughout the playbook. Variables defined on the step level are available in that step, and in any step that executes in a sub-branch of an `if-condition`, `while-condition`, or `parallel` step.
+
+According to the CACAO spec, variable names should start and end with double underscores (`__`), but this is not enforced by SOARCA. Furthermore, the CACAO spec allows defining multiple types of variables (strings, ip-addresses, numbers), but at this time SOARCA will interpret every value as a string. The `constant` and `external` fields are also ignored.
+
+```json
+"playbook_variables": {
+    "__target_ip__": {
+      "type": "ipv4-addr",
+      "description": "IP address of target system",
+      "constant": false,
+      "external": true
+    }
+}
+```
+
+SOARCA supports the interpolation of variables in different strings. The specific string-based fields that support interpolation depend on the capability. In general, string interpolation is supported in the fields of agents, targets, and `command` fields.
+
+Variable interpolation happens at the last possible moment, which means that step-dependant variables can be used in agent and target definitions.
+
+Substitution is performed by replacing any occurrence of `[variable_name]:value` with the string `value` of that variable. Undefined variables are not replaced.
+
+### Action steps
+
+Within CACAO playbooks, `action` steps can define commands that are executed by an _agent_ against one or more _targets_. The agent and targets are referenced by ID. SOARCA selects the internal capability for handling the step by looking at the `type` and `name` of the agent. After selecting the proper capability, SOARCA will sequentially perform every command in the `commands` field for every target specified in `targets`. If any command fails to execute successfully, further execution is halted and the step is considered to have failed.
+
+Action steps may return variables. On the successful execution of an action step, any variables returned are added to the globally available playbook variables. If the `out_args` field is specified and non-empty, only the variables listed in `out_args` will be added to the global playbook variables. The `in_args` field from the CACAO spec is ignored. Any variable defined on the playbook level, in parent-steps and within the step itself are available for interpolation.
```

Comment:
Let's just add a note to explicitly mention that this is how SOARCA handles variables.